src/db.c

```diff
@@ -2010,21 +1982,22 @@ static int _db_rule_gen_arg_32(struct db_sys_list *s_new,
 }
 
 /**
- * Generate a new filter rule for a 32 bit system
+ * Generate a new filter rule
  * @param arch the architecture definition
  * @param rule the new filter rule
  *
- * This function generates a new syscall filter for a 32 bit system. Returns
- * zero on success, negative values on failure.
+ * This function generates a new syscall filter. Returns
+ * an allocated db_sys_struct on success, NULL on failure.
  *
  */
-static struct db_sys_list *_db_rule_gen_32(const struct arch_def *arch,
-					   const struct db_api_rule_list *rule)
+static struct db_sys_list *_db_rule_gen(const struct arch_def *arch,
+					const struct db_api_rule_list *rule)
 {
 	unsigned int iter;
 	struct db_sys_list *s_new;
 	const struct db_api_arg *chain = rule->args;
 	struct db_arg_chain_tree *prev_nodes[4] = { NULL, };
+	bool is_32bit = arch->size == ARCH_SIZE_32;
```

Comment:
Since `is_32bit` is only used once below, why not just drop the dedicated local variable and just do the `arch->size` comparison where needed?